test/unit/org/apache/cassandra/db/filter/ColumnFilterTest.java

```diff
@@ -18,53 +18,348 @@
 
 package org.apache.cassandra.db.filter;
 
+import java.io.IOException;
+import java.util.function.Consumer;
+
+import com.google.common.base.Throwables;
+import org.junit.Assert;
 import org.junit.Test;
 
-import junit.framework.Assert;
 import org.apache.cassandra.config.CFMetaData;
 import org.apache.cassandra.config.ColumnDefinition;
+import org.apache.cassandra.db.PartitionColumns;
 import org.apache.cassandra.db.marshal.Int32Type;
+import org.apache.cassandra.db.marshal.SetType;
+import org.apache.cassandra.db.rows.CellPath;
 import org.apache.cassandra.dht.Murmur3Partitioner;
 import org.apache.cassandra.io.util.DataInputBuffer;
 import org.apache.cassandra.io.util.DataInputPlus;
 import org.apache.cassandra.io.util.DataOutputBuffer;
 import org.apache.cassandra.net.MessagingService;
 import org.apache.cassandra.utils.ByteBufferUtil;
 
+import static org.junit.Assert.assertEquals;
+
 public class ColumnFilterTest
 {
-    final static ColumnFilter.Serializer serializer = new ColumnFilter.Serializer();
+    private static final ColumnFilter.Serializer serializer = new ColumnFilter.Serializer();
+
+    private final CFMetaData metadata = CFMetaData.Builder.create("ks", "table")
+                                                          .withPartitioner(Murmur3Partitioner.instance)
+                                                          .addPartitionKey("pk", Int32Type.instance)
+                                                          .addClusteringColumn("ck", Int32Type.instance)
+                                                          .addStaticColumn("s1", Int32Type.instance)
+                                                          .addStaticColumn("s2", SetType.getInstance(Int32Type.instance, true))
+                                                          .addRegularColumn("v1", Int32Type.instance)
+                                                          .addRegularColumn("v2", SetType.getInstance(Int32Type.instance, true))
+                                                          .build();
+
+    private final ColumnDefinition s1 = metadata.getColumnDefinition(ByteBufferUtil.bytes("s1"));
+    private final ColumnDefinition s2 = metadata.getColumnDefinition(ByteBufferUtil.bytes("s2"));
+    private final ColumnDefinition v1 = metadata.getColumnDefinition(ByteBufferUtil.bytes("v1"));
+    private final ColumnDefinition v2 = metadata.getColumnDefinition(ByteBufferUtil.bytes("v2"));
+    private final CellPath path0 = CellPath.create(ByteBufferUtil.bytes(0));
+    private final CellPath path1 = CellPath.create(ByteBufferUtil.bytes(1));
+    private final CellPath path2 = CellPath.create(ByteBufferUtil.bytes(2));
+    private final CellPath path3 = CellPath.create(ByteBufferUtil.bytes(3));
+    private final CellPath path4 = CellPath.create(ByteBufferUtil.bytes(4));
+
+    // Select all
+
+    @Test
+    public void testSelectAll() throws Exception
```

Comment:
Nit: Now that `testRoundTrip` body in inside a try-catch block (I assume that to avoid problems when calling it from the lambdas), none of the methods need the `throws Exception` part.